memory/vatest.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2017 IBM
+# Author:Praveen K Pandey<praveen@linux.vnet.ibm.com>
+#
+
+import os
+import shutil
+
+from avocado import Test
+from avocado import main
+from avocado.utils import process,  build, memory
+from avocado.utils.software_manager import SoftwareManager
+
+
+class VATest(Test):
+    """
+    Performs Virtual address space validation
+
+    :avocado: tags=memory, power
+    """
+
+    def setUp(self):
+        '''
+        Build VA Test
+        '''
+
+        # Check for basic utilities
+        smm = SoftwareManager()
+        self.scenario_arg = self.params.get('scenario_arg', default='1')
+
+        if self.scenario_arg in [2, 3, 4]:
+            if memory.meminfo.Hugepagesize.mb != 16:
+                self.cancel(
+                    "Test need to skip as 16MB huge need to configured")
+            if self.scenario_arg == 4:
+                memory.set_num_huge_pages(131072)
+            else:
+                memory.set_num_huge_pages(1024)
+        elif self.scenario_arg in [5, 6, 7]:
+            if memory.meminfo.Hugepagesize.gb != 16:
+                self.cancel(
+                    "Test need to skip as 16GB huge need to configured")
+            if self.scenario_arg == 7:
+                memory.set_num_huge_pages(2)
+            else:
+                memory.set_num_huge_pages(1)
```

Comment:
What will the test do if `scenario_arg` is a value not handled ?

Can it be like:
```
if memory.meminfo.Hugepagesize.mb != 16:
    self.cancel()
dic = {'2': 1024, '3', 1024, ...}
if self.scenario_arg in dic:
    memory.set_num_huge_pages(dic[self.scenario_arg])
```

Reply:
yes make sense , thanks